### DISCUSSION TITLE
Multiple Rx and one Tx in Sionna RT Stream Management

### DISCUSSION THREAD

**Discussion by pablosreyero** [NVlabs/sionna#324] [Q&A] [answered]
Hi,

I was wondering if Sionna OFDM Stream Management function, supports multiple receivers in a scene with just one Tx. I am asking this, due to the fact that with the following code I do not obtain the indexes of all Rxs, and therefore, most of `stream_management` properties are  incorrect. 

Here the code: 
[OFDM_MIMO_RT_code.zip](https://github.com/NVlabs/sionna/files/14124949/OFDM_MIMO_RT_code.zip)

Here the `stream_management` properties I obtain:

```
rx_tx_association = np.zeros([num_Rx, num_Tx])

# rx_tx_association[0, 0] = 1
# rx_tx_association[1, 0] = 1
# rx_tx_association[2, 0] = 1
# rx_tx_association[3, 0] = 1
# rx_tx_association[4, 0] = 1

rx_tx_association[:,0] =  1 # lo tenemos que hacer asi

#----------------RESULTS----------------#

'''
Here the number of streams per tx:  1 # this one is correct
Here the number of streams per rx:  0 # this one should one
Here the IDs:
 [[0]] # this array should be populated
[]
This is the number of rx communicating with a tx:  5 # correct
This is the number of interfering streams per tx:  1 # i'm not sure about this one
'''
```

**Comment by jhoydis** [comment]
Hi, 

We'll have a look and come back to you quickly.

**Comment by jhoydis** [comment, marked as answer]
Hi,

I had a quick look at your code and there are multiple issues:

1) You want to have one TX and 5 RX. In this case, you must set `num_streams_per_tx=5` because the TX is sending 5 streams, one for each RX.

2) You cannot send 5 streams with a single TX antenna. The number of streams must be smaller than or equal to the number of TX antennas. 

3) The TX would probably need to apply a form of precoding.

Here is a code snippet, that shows the correct setting:
```
num_tx = 1
num_rx = 5
num_streams_per_tx = num_rx
rx_tx_association = np.zeros([num_rx, num_tx])
rx_tx_association[:,0] =  1 
sm = StreamManagement(rx_tx_association, num_streams_per_tx)
print('Number of streams per tx: ', sm.num_streams_per_tx)
print('Number of streams per rx: ', sm.num_streams_per_rx)
print('Stream IDs:\n', sm.tx_stream_ids)
print('Number of rx communicating with a tx: ', sm.num_rx_per_tx)
print('Number of interfering streams per rx: ', sm.num_interfering_streams_per_rx)
```
```console
Number of streams per tx:  5
Number of streams per rx:  1
Stream IDs:
 [[0 1 2 3 4]]
Number of rx communicating with a tx:  5
Number of interfering streams per rx:  4
```

**Reply by pablosreyero** [reply]
Hi Jakob,

Thanks for your clear response, I'm a bit new to this world, so I misunderstood some concepts and while implementing my code, it did not make that much sense. Everything is clear now.

Cheers,

Pablo.-
